guides/messaging-high-availability.adoc

```diff
@@ -0,0 +1,228 @@
+= Configure WildFly With Messaging (ActiveMQ Artemis) Cluster And High Availability
+:summary: Learn how to configure WildFly with messaging (ActiveMQ Artemis) cluster and high availability
+:includedir: _includes
+include::{includedir}/_attributes.adoc[]
+// you can override any attributes eg to lengthen the
+// time to complete the guide
+:prerequisites-time: 20
+
+In this guide you will learn how to configure two WildFly servers with messaging (ActiveMQ Artemis broker) in high availability topology with shared journal.
+
+Apache ActiveMQ Artemis broker is integrated in WildFly and allows Artemis brokers to be linked together in primary-secondary pairs.
+It is active-passive high availability design such that one (the primary node) handles all the workload under normal operating conditions,
+while the other (the secondary node) remains on standby, ready to take over if the active node fails. As usually it’s not desirable to keep
+one Wildfly server idle on standby mode, we will configure one additional ActiveMQ Artemis server on each WidFly server as a secondary for the other.
+So there will be two primary-secondary pairs, each pair with its own journal directory.
+
+image::artemis-collocated-topology.png[WildFly with ActiveMQ Artemis in collocated HA topology]
+
+If one WildFly server fails, the secondary ActiveMQ Artemis broker on the other WildFly server will activate and take over all duties.
+All HA enabled connections from the original primary broker will fail over to the secondary broker. The secondary broker will
+access all messages stored on the primary broker by loading them from the shared journal directory. Once the original WildFly server is restarted,
+the secondary broker will automatically switch back to standby mode, and all clients will fail back to the primary broker.
+Something like:
+
+[NOTE]
+--
+For simplicity, we will use just one machine to run WildFly instances and local directories for shared journals. In real world scenario those should be two separate machines with shared journal mount from shared file system (like NFS4).
+--
+
+[[prerequisites]]
+== Prerequisites
+
+To complete this guide, you need:
+
+* Roughly {prerequisites-time} minutes
+* JDK {jdk-minimal-version} installed with `JAVA_HOME` configured appropriately
+
+== Prepare shared journal directories
+
+Create two directories for each primary-secondary pair:
+
+[source, bash ,options="nowrap"]
+----
+mkdir messaging-journal-a
+mkdir messaging-journal-b
+----
+
+== Prepare WildFly Servers
+
+Now there will be configured two WildFly servers with two primary-secondary pairs. ActiveMQ Artemis configured
```

Comment:
Now let's configure two WildFly servers with two primary-secondary pairs of embedded ActiveMQ Artemis servers.

Reply:
Fixed.